src/components/dialog/dialog.tsx

```diff
@@ -114,7 +116,7 @@ export class Dialog {
             : '';
     }
 
-    public componentDidUnload() {
+    public disconnectedCallback() {
         this.mdcDialog.unlisten('MDCDialog:opened', this.handleMdcOpened);
         this.mdcDialog.unlisten('MDCDialog:closed', this.handleMdcClosed);
         this.mdcDialog.unlisten('MDCDialog:closing', this.handleMdcClosing);
```

Comment:
Again, we probably need to run the setup on `connectedCallback`.

Reply:
One note about this is that `connectedCallback` will run before the first render and then we will not be able to setup the mdc components. We might have to use `componentDidRender` instead and just set a flag in `connectedCallback` if we should initialize the mdc components or not :thinking: